### ISSUE TITLE
Latest version of Angular installs tapable 2.2.0 which conflicts with storybook 1.1.3 requirement

### ISSUE THREAD

**Issue by themizzi** [storybookjs/storybook#13440]
**Describe the bug**
After installing storybook into a clean angular 11.0.4 project, tapable throws a number of errors while compiling typescript.

**To Reproduce**
1. Create a fresh Angular 11.0.4 project.
2. Init storybook.
3. Run tsc.

**Expected behavior**
tsc compiles without errors

**Screenshots**

**Code snippets**
results:

```
node_modules/@types/webpack/index.d.ts:32:3 - error TS2305: Module '"../../tapable/tapable"' has no exported member 'Tapable'.

32   Tapable,
     ~~~~~~~

node_modules/@types/webpack/index.d.ts:1062:23 - error TS2707: Generic type 'SyncWaterfallHook<T, AdditionalOptions>' requires between 1 and 2 type arguments.

1062             resolver: SyncWaterfallHook;
                           ~~~~~~~~~~~~~~~~~
```
_and many more_

**System**
```
Environment Info:

  System:
    OS: macOS 11.0.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  Binaries:
    Node: 12.15.0 - ~/.nvm/versions/node/v12.15.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v12.15.0/bin/yarn
    npm: 6.14.9 - ~/.nvm/versions/node/v12.15.0/bin/npm
  Browsers:
    Chrome: 87.0.4280.88
    Edge: 87.0.664.60
    Firefox: 80.0.1
    Safari: 14.0.1
  npmPackages:
    @storybook/addon-actions: ^6.1.11 => 6.1.11 
    @storybook/addon-essentials: ^6.1.11 => 6.1.11 
    @storybook/addon-links: ^6.1.11 => 6.1.11 
    @storybook/angular: ^6.1.11 => 6.1.11 
```

**Additional context**

I was able to resolve by forcing tsc 1.1.3, but unfortunately that will likely cause issues with angular dev kit which transiently requires 2.2.0.

```
angular-to-do@0.0.0
├─┬ @angular-devkit/build-angular@0.1100.4
│ ├─┬ @ngtools/webpack@11.0.4
│ │ └─┬ enhanced-resolve@5.3.1
│ │   └── tapable@2.2.0 
│ └─┬ webpack@4.44.2
│   ├─┬ enhanced-resolve@4.3.0
│   │ └── tapable@1.1.3  deduped
│   └── tapable@1.1.3 
└─┬ @storybook/angular@6.1.11
  ├─┬ @storybook/core@6.1.11
  │ ├─┬ html-webpack-plugin@4.5.0
  │ │ └── tapable@1.1.3 
  │ └─┬ react-dev-utils@10.2.1
  │   └─┬ fork-ts-checker-webpack-plugin@3.1.1
  │     └── tapable@1.1.3 
  ├─┬ fork-ts-checker-webpack-plugin@4.1.6
  │ └── tapable@1.1.3 
  ├─┬ ts-loader@6.2.2
  │ └─┬ enhanced-resolve@4.3.0
  │   └── tapable@1.1.3 
  └─┬ tsconfig-paths-webpack-plugin@3.3.0
    └─┬ enhanced-resolve@4.3.0
      └── tapable@1.1.3 
```

**Comment by shilman** [comment]
@storybookjs/angular has anybody seen this and/or know the workaround?

**Comment by chriscarpenter12** [comment]
Seeing the exact same in a project I'm upgrading to Angular 11 and Storybook 6.1.10

**Comment by meriturva** [comment]
Same issue here with Angular 11.0.9 and StoryBook 6.1.14 (no workaround found)

**Comment by chriscarpenter12** [comment]
Like @themizzi described in his initial post using a for resolution I was able to get my library to build. Not sure what will break also.

`package.json` snippet to get it working
```json
{
  "scripts": {
    "preinstall": "npx npm-force-resolutions"
  },
  "resolutions": {
    "tapable": "1.1.3"
  }
}
```

**Comment by Imp0ssible** [comment]
`"compilerOptions": {
		"skipLibCheck": true
	}`

**Comment by meriturva** [comment]
Sorry @Imp0ssible but this a problem of StoryBook. Enabling skipLibCheck is not a valid solution for many projects.

**Comment by toamitc** [comment]
Issue exist with version 6.1.17 as well. This is storybug issue and skipLibCheck is not correct way.  Any solution for this?

**Comment by deepakanuraags** [comment]
Just wanted to know if this issue will be resolved and when can it be expected to be resolved.
I used the package.json snippet given by @chriscarpenter12 and it seems to work fine. However, as said by him we do not know what might break if we fix the tapable version to 1.1.3

**Comment by ThibaudAV** [comment]
if you can add a repo with the reproduction of the problem 🥺 🙏 ? 
it could be faster to help and see what's going on 

Je n'ai pas réussi a reproduire exactement ce problème

**Comment by deepakanuraags** [comment]
@ThibaudAV i unfortunately cannot do that, however the problem i am currently facing is that i am able to build locally but in pipeline under while npm resolution comes it fails because of timeout

**Comment by eljay040** [comment]
`npm audit fix` did the fix for me (Angular 11, Storybook 6.3.4)

**Comment by i5irin** [comment]
I am in a similar situation.
[This is a repository](https://github.com/i5irin-Reproduction/storybook-webpack-conflict) for situations where type errors occur during builds.
It can be reproduced by entering the container with VSCode and "npm run build".

**Comment by LeLunZ** [comment]
@shilman this is still an issue.
Can't get storybook to work with the latest angular@13 version.

will there be a solution?

**Comment by chrisbraddock** [comment]
Same w/ Angular 14 and @angular-builders/custom-webpack.

**Comment by ndelangen** [comment]
Should be fixed in 7.0 beta.
